### PR TITLE
Incorporate feedback from HEs in the GitHub issue template.

### PR DESCRIPTION
**Pull request by worldomonation**
#### Changes proposed in this Pull Request

This PR modifies the GitHub issue template after feedback from HEs.

Key changes:

- several fields are no longer required (Browser, OS, Context).
- OS version field removed.
- Summary field added immediately below the welcome text.
- Notes/Error log fields merged into one.

Threads: p1628894017128700-slack-C01UW7SB411 and p1629133543059100/1628098234.020300-slack-C02FMH4G8

#### Testing instructions

Browse around [here](https://github.com/Automattic/wp-calypso/blob/update/github-issue-template/.github/ISSUE_TEMPLATE/happiness-bug-report.yml).

Related to #

### CODE REVIEW

**Review by dpasque — APPROVED** [review]
Looks good from Q-squad perspective, but I would wait for HE approval before merge!

**Comment by inaikem** [comment]
Thanks @dpasque, we have had a fair bit of renewed discussion on this over the past few days, so expect a few more tweaks soon! 🙏 

Discussion: pcLjiI-i3-p2 + pcLjiI-i3-p2#comment-288

**Pull request merged**
